File: WebAPI/Models/AvailableTimeModel.cs
using System;

namespace WebAPI.Models
{
    public class AvailableTimeModel
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public int AdvertId { get; set; }
        public int AvailabilityStateId { get; set; }
    }
}

File: WebAPI/ClientApp/src/app/models/AvailableTimeModel.ts
export class AvailableTimeModel {
    public id: number;
    public date: Date;
    public advertId: number;
    public availabilityStateId: number;

    public constructor(_id: number, _date: Date, _advertId: number, _availabilityStateId: number) {
        this.id = _id;
        this.date = _date;
        this.advertId = _advertId;
        this.availabilityStateId = _availabilityStateId;
    }
}